### PR TITLE
scripts/pythondistdeps: Only print rich dep list when required to.

### PR DESCRIPTION
**Pull request by gordonmessmer**

### CODE REVIEW

**Comment by gordonmessmer** [comment]
This should resolve #981

File: scripts/pythondistdeps.py [hunk 1]
```diff
@@ -269,7 +269,10 @@
                 spec_list.append('{n} >= {v} with {n} < {vnext}'.format(n=name, v=spec[1], vnext=next_ver))
             else:
                 spec_list.append('{} {} {}'.format(name, spec[0], spec[1]))
-        print('(%s)' % ' with '.join(spec_list))
+        if len(spec_list) == 1:
+            print(spec_list[0])
+        else:
+            print('(%s)' % ' with '.join(spec_list))
```
Comment (ignatenkobrain) [inline scripts/pythondistdeps.py]: can we keep same style of code?

```
print('({})'.format(' with '.join(spec_list)))
```

?

**Comment by gordonmessmer** [comment]
```
[gordon@vagabond:~/git/rpm]$ find /home/gordon/git/molecule-root/ | python3 ~/git/rpm/scripts/pythondistdeps.py -R
python(abi) == 3.7
python3.7dist(ansible) >= 2.5
(python3.7dist(ansible-lint) >= 4.1.1a2 with python3.7dist(ansible-lint) < 5)
python3.7dist(cerberus) >= 1.3.1
python3.7dist(click) >= 7
python3.7dist(click-completion) >= 0.5.1
python3.7dist(colorama) >= 0.3.9
python3.7dist(cookiecutter) >= 1.6
python3.7dist(flake8) >= 3.6
python3.7dist(jinja2) >= 2.10.1
(python3.7dist(paramiko) < 3 with python3.7dist(paramiko) >= 2.5)
(python3.7dist(pexpect) >= 4.6 with python3.7dist(pexpect) < 5)
(python3.7dist(pluggy) >= 0.7.1 with python3.7dist(pluggy) < 1)
(python3.7dist(pre-commit) < 2 with python3.7dist(pre-commit) >= 1.17)
(python3.7dist(python-gilt) < 2 with python3.7dist(python-gilt) >= 1.2.1)
(python3.7dist(pyyaml) >= 5.1 with python3.7dist(pyyaml) < 6)
python3.7dist(setuptools)
python3.7dist(sh) >= 1.12.14
python3.7dist(six) >= 1.11
python3.7dist(tabulate) >= 0.8.4
python3.7dist(testbar) >= 2.1.2 with python3.7dist(testbar) < 2.2
python3.7dist(testbaz) >= 3.2.1 with python3.7dist(testbaz) < 3.2.2
python3.7dist(testfoo) < 1.0.1 or python3.7dist(testfoo) >= 1.0.1.0
(python3.7dist(testinfra) < 4 with python3.7dist(testinfra) >= 3.0.6)
python3.7dist(tree-format) >= 0.1.2
(python3.7dist(yamllint) < 2 with python3.7dist(yamllint) >= 1.15)

[gordon@vagabond:~/git/rpm]$ find /home/gordon/git/molecule-root/ | python3 ~/git/rpm/scripts/pythondistdeps.py -P
python3.7dist(molecule) == 3.0a5.dev15+gec8c85e9.d20191124
```

File: scripts/pythondistdeps.py (hunk 1 above)
Comment (gordonmessmer) [reply scripts/pythondistdeps.py]: I've updated the branch.  Thank you.

**Comment by ignatenkobrain** [comment]
> python3.7dist(molecule) == 3.0a5.dev15+gec8c85e9.d20191124

It would be very cool if we could actually mangle this to a proper RPM notation, but this is definitely outside of the scope of this PR.

**Comment by Conan-Kudo** [comment]
@gordonmessmer You beat me to it. Thanks for this! 👍

**Comment by gordonmessmer** [comment]
@Conan-Kudo Seems only right that I fix what I broke.  :)

**Comment by gordonmessmer** [comment]
@ignatenkobrain I started a repo to flesh out some of the requirement conversions from python dist tools to rpm.  It doesn't mangle versions right now, but maybe you could open an issue there and describe what should happen?

https://github.com/gordonmessmer/pyreq2rpm/

**Comment by hroncok** [comment]
Given `pip~=19.3`, this would generate:

```
python3.8dist(pip) >= 19.3
with
python3.8dist(pip) < 20
```

However, https://github.com/rpm-software-management/rpm/pull/996 fixes that.

**Pull request merged**